### PR TITLE
Rework python build system to scikit-build-core

### PR DESCRIPTION
**Pull request by dudoslav**
This PR introduces a new way of packaging Python library using scikit-build-core.

Main change had to be to move the pyproject.toml to the top-level folder.

For now we have to include .git in the sdist package, since the src/src/CMakeLists.txt uses git to query version.
In the future we would like to define these values during packaging process.

Also: This PR concerns tasks 38316 and 38319 as well

### CODE REVIEW

File: src/cmake/Superbuild.cmake [hunk 1]
```diff
@@ -131,12 +132,22 @@ ExternalProject_Add(libtiledbvectorsearch
 # make install-libtiledbvectorsearch
 add_custom_target(install-libtiledbvectorsearch
   COMMAND
-    ${CMAKE_COMMAND} --build . --target install --config ${CMAKE_BUILD_TYPE}
+    ${CMAKE_COMMAND} --build . --target install
```
Comment (teo-tsirpanis) [inline src/cmake/Superbuild.cmake:135]:
```suggestion
    ${CMAKE_COMMAND} --build . --target install --config $<CONFIG>
```

File: src/cmake/Superbuild.cmake [hunk 2]
```diff
@@ -131,12 +132,22 @@ ExternalProject_Add(libtiledbvectorsearch
...
   COMMAND
     ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_BINARY_DIR}/libtiledbvectorsearch/install_manifest.txt ${CMAKE_CURRENT_BINARY_DIR}/install_manifest.txt
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/libtiledbvectorsearch
 )
 
+add_custom_target(
+        skbuild-export
+        COMMAND
+            ${CMAKE_COMMAND} --install . --prefix ${SKBUILD_PLATLIB_DIR}/tiledb/vector_search
```
Comment (teo-tsirpanis) [inline src/cmake/Superbuild.cmake:144]:
```suggestion
            ${CMAKE_COMMAND} --install . --prefix ${SKBUILD_PLATLIB_DIR}/tiledb/vector_search --config $<CONFIG>
```

File: pyproject.toml [hunk 3]
```diff
@@ -27,34 +27,38 @@ dependencies = [
...
-    "scikit-build>=0.13", "pybind11[global]>=2.10",
-    "cmake"
+requires = ["scikit-build-core", "pybind11"]
+build-backend = "scikit_build_core.build"
+
+[tool.scikit-build]
+wheel.expand-macos-universal-tags = true
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+cmake.verbose = true # TODO: Remove
+logging.level = "INFO" # TODO: Remove
```
Comment (teo-tsirpanis) [inline pyproject.toml:42]:
```suggestion
```

**Review by teo-tsirpanis — APPROVED** [review]
Mostly fine, let's wait for @NikolaosPapailiou if he has any comments.

**Review by NikolaosPapailiou — APPROVED** [review]
Lets also wait for @ihnorton to approve

File: .github/workflows/ci_python.yml [hunk 4]
```diff
@@ -28,8 +28,8 @@ jobs:
           python --version
       - name: Build and test python
         run: |
-          cd apis/python
           pip install .[test]
+          cd apis/python
           pytest
```
[the remark this answers is not in the document]
Comment (dudoslav) [reply .github/workflows/ci_python.yml:33]: On it

[the remark this answers is not in the document]
Comment (dudoslav) [reply .github/workflows/ci_python.yml:33]: Fixed, hopefully I found all references to building python package. I also changed dockerfile.

[the remark this answers is not in the document]
Comment (jparismorgan) [reply .github/workflows/ci_python.yml:33]: Awesome, thank you! Do we also need update these?

> To test Demo notebooks:
> ```bash
> cd apis/python
> pip install -r test/ipynb/requirements.txt
> pytest --nbmake test/ipynb
> ```
> 
and

> You can run unit tests with `pytest`. You'll also need to install the test dependencies:
> ```bash
> cd apis/python
> pip install ".[test]"
> ```
> Then you can run the tests:
> ```bash
> cd apis/python
> # To run all tests.
> pytest
> # To run a single test and display standard output and standard error.
> pytest test/test_ingestion.py -s
> ```

[the remark this answers is not in the document]
Comment (dudoslav) [reply .github/workflows/ci_python.yml:33]: Where did you find those?

[the remark this answers is not in the document]
Comment (dudoslav) [reply .github/workflows/ci_python.yml:33]: Ah I see

[the remark this answers is not in the document]
Comment (dudoslav) [reply .github/workflows/ci_python.yml:33]: In the order you send them:

1. Looking at this snippet if it is in top level README.md I believe it was incorrect even before my changes. Will fix.
2. cant find
3. cant find

File: README.md [hunk 5]
```diff
@@ -40,4 +40,17 @@ We welcome contributions. Please see [`Building`](./documentation/Building.md) f
...
+* Unit tests: `pytest`
+* Demo notebooks:
+  * ```
+    cd apis/python
+    pip install -r test/ipynb/requirements.txt
+    pytest --nbmake test/ipynb
+    ```
+* Credentials:
+  * Some tests run on TileDB Cloud using your current environment variable `TILEDB_REST_TOKEN` -- you will need a valid API token for the tests to pass 
+  * For continuous integration, the token is configured for the `unittest` user and all tests should pass
```
Comment (jparismorgan) [inline README.md:56]: (Minor) I don't think we should add this back? I moved it to `Building.md` in this PR https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/192/files - though if you want it in `README.md` instead I'm fine with that, but let's remove it `Building.md` then.

Comment (dudoslav) [reply README.md:56]: Ah yea, since this PR got postponed I had to manually merge some changes

**Comment by dudoslav** [comment]
Thank you @jparismorgan for checking the documentation.

File: README.md (hunk 5 above)
Comment (jparismorgan) [reply README.md]: No problem sounds good, thanks for updating

**Comment by dudoslav** [comment]
Postponed after next release

**Pull request merged**